AutomatedLabWorker/AutomatedLabVMWareWorkerVirtualMachines.psm1

```diff
@@ -1,11 +1,53 @@
+<# 
+    An AutomatedLab user could have a VMWare environment on premisis _and_ have Hyper-V enabled on his own device.
+    When the user has both Hyper-V and VMWare modules loaded in one session, this can cause unwanted behaviour.
+    
+    This may need to be mitigated in AutomatedLab in some cases. There are two main approaches:
+    1) Prepending conflicting CmdLets with the name of the intende module of origin, e.g.:
+        VMware.VimAutomatiion.Core\Get-VM 
+
+    2) Theoretically, one should be able to load modules using the -Prefix parameter:
+        Import-Module VMWare* -Prefix "VW"
+    This _should_ result in all the VMWare CmdLets being imported with the prefix VW:
+        Get-VWVM 
+    Unfortunately, this does not work in PowerCLI 6.0R2 - 6.5.0 , because of the way the underlying PSSnapin is loaded.
+    See here for more information: 
+        https://communities.vmware.com/thread/520601
+    This will be solved in PowerCLI version 6.5.1 R1:
+        https://blogs.vmware.com/PowerCLI/2016/11/saying-farewell-snapins.html
+
+    For now, approach 1) is probably the way to go, or we should force users to require version 6.5.1R1 or greater for the VMware module.
+    
+    Get a list of CmdLets whith the same name in Hyper-V and VMWare:
+    Compare-Object (get-command -Module vmware.vimautomation.core) (Get-Command -Module Hyper-v) -IncludeEqual -ExcludeDifferent
+    Retrieved 30-5-2017, Vmware module version:6.3.0.0, Hyper-V module version 2.0.0.0
+
+        InputObject SideIndicator
+        ----------- -------------
+        Get-VM      ==
+        Get-VMHost  ==
+        Move-VM     ==
+        New-VM      ==
+        Remove-VM   ==
+        Restart-VM  ==
+        Set-VM      ==
+        Set-VMHost  ==
+        Start-VM    ==
+        Stop-VM     ==
+        Suspend-VM  ==
+#>
+
 trap
 {
     if ((($_.Exception.Message -like '*Get-VM*') -or `
          ($_.Exception.Message -like '*Save-VM*') -or `
          ($_.Exception.Message -like '*Get-VMSnapshot*') -or `
          ($_.Exception.Message -like '*Suspend-VM*') -or `
-         ($_.Exception.Message -like '*CheckPoint-VM*')) -and (-not (Get-Module -ListAvailable Hyper-V)))
+         ($_.Exception.Message -like '*CheckPoint-VM*')) -and `
+         (-not (Get-Module -ListAvailable Hyper-V))) 
     {
+        # What is the exact purpose of this error trap? 
```

Comment:
Thanks @PerPedersenMSFT, @raandree for getting back to me on this. I'll feel free to fiddle with them and run some tests, and maybe the lines are no longer neede then.